packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -617,6 +620,45 @@ export class Cluster extends Resource implements ICluster {
     return this._awsAuth;
   }
 
+  /**
+   * If this cluster is kubectl-enabled, returns the OpenID Connect issuer url.
+   * This is because the values is only be retrieved by the API and not exposed
+   * by CloudFormation. If this cluster is not kubectl-enabled (i.e. uses the
+   * stock `CfnCluster`), this is `undefined`.
+   * @attribute
+   */
+  public get clusterOpenIdConnectIssuerUrl(): string {
+    if (!this._clusterResource) {
+      throw new Error('unable to obtain OpenID Connect issuer URL. Cluster must be kubectl-enabled');
+    }
+
+    return this._clusterResource.attrOpenIdConnectIssuerUrl;
+  }
+
+  /**
+   * Lazily creates the OpenIdConnectProvider resource, which links the cluster to AWS IAM.
+   */
```

Comment:
Focus on what this is from a user perspective and less about the implementation:

```suggestion
  /**
   * An `OpenIdConnectProvider` resource associated with this cluster, and which can be used 
   * to link this cluster to AWS IAM.
   * 
   * A provider will only be defined if this property is accessed (lazy initialization).
   */
```